include/cuco/detail/dynamic_map.inl

```diff
@@ -128,34 +171,85 @@ void dynamic_map<Key, Value, Scope, Allocator>::insert(InputIt first,
   }
 }
 
+template <typename Key, typename Value, cuda::thread_scope Scope, typename Allocator>
+template <typename InputIt, typename Hash, typename KeyEqual>
+void dynamic_map<Key, Value, Scope, Allocator>::erase(
+  InputIt first, InputIt last, Hash hash, KeyEqual key_equal, cudaStream_t stream)
+{
+  std::size_t num_keys = std::distance(first, last);
+
+  auto const block_size = 128;
+  auto const stride     = 1;
+  auto const tile_size  = 4;
+  auto const grid_size  = (tile_size * num_keys + stride * block_size - 1) / (stride * block_size);
+
+  // TODO: memset an atomic variable is unsafe
+  static_assert(sizeof(std::size_t) == sizeof(atomic_ctr_type));
+
+  // zero out submap success counters
+  static_assert(sizeof(std::size_t) == sizeof(atomic_ctr_type));
```

Comment:
Duplicate check.
Can we also emit a suitable error message?

Reply:
Yep, added that in.